proxy/http/HttpTransact.cc

```diff
@@ -5896,6 +5932,17 @@ HttpTransact::is_cache_response_returnable(State *s)
     SET_VIA_STRING(VIA_DETAIL_CACHE_LOOKUP, VIA_DETAIL_MISS_METHOD);
     return false;
   }
+  // If we are caching responses to POST, make sure that our cached resource
+  // has a method that matches the incoming requests's method. If not, then we
+  // cannot reply with the cached resource. That is, we cannot reply to an
+  // incoming GET request with a response to a previous POST request.
+  int const client_request_method = s->hdr_info.client_request.method_get_wksidx();
+  int const cached_request_method = s->cache_info.object_read->request_get()->method_get_wksidx();
+  if (s->http_config_param->cache_post_method == 1 && client_request_method != cached_request_method) {
```

Comment:
Why check the config params? Are there cases where the item should be served even if the methods don't match?

Reply:
Yeah, good point. There probably isn't a good reason to return a resource for a mismatched method. I added the config check because it reduced the scope of the impact of my change, reducing the risk of this patch. But I think it should be safe to remove.